### PR TITLE
Fix "Called C++ object pointer is null" in ApvAnalysisFactory::constructAuxiliaryApvClasses

### PR DESCRIPTION
**Pull request by iarspider**
#### PR description:

LLVM analyzer [reports](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_14_2_X_2024-09-16-1100/el8_amd64_gcc12/llvm-analysis/report-818127.html#EndPath) "Called C++ object pointer is null" in `ApvAnalysisFactory::constructAuxiliaryApvClasses`. This PR adds early return statements to avoid this.

#### PR validation:

Bot tests

### CODE REVIEW

**Comment by iarspider** [comment]
please test

File: CalibTracker/SiStripAPVAnalysis/src/ApvAnalysisFactory.cc [hunk 1]
```diff
@@ -118,7 +118,10 @@ void ApvAnalysisFactory::constructAuxiliaryApvClasses(ApvAnalysis* theAPV, uint3
       theCM = new MedianCommonModeCalculator();
     } else {
       cout << "Sorry Only Median is available for now, Mean and FastLinear are coming soon" << endl;
+      return;
     }
+  } else {
+    return;
```
Comment (perrotta) [inline CalibTracker/SiStripAPVAnalysis/src/ApvAnalysisFactory.cc:124]: If one returns here, probably better not doing it silently. A `cout`, as in previous [L120](https://github.com/cms-sw/cmssw/pull/46074/files#diff-c191c0b9b0798dc37092e887ee8affa0ee4a0f774d4a96e1d6925caaa9e03fcdR120), could work:
```suggestion
  } else {
    cout << "ApvAnalysisFactory: algorithm " << theAlgorithmType_ << " not supported" << endl:
    return;
```

**Comment by iarspider** [comment]
please test

File: CalibTracker/SiStripAPVAnalysis/src/ApvAnalysisFactory.cc [hunk 2]
```diff
@@ -118,7 +118,11 @@ void ApvAnalysisFactory::constructAuxiliaryApvClasses(ApvAnalysis* theAPV, uint3
       theCM = new MedianCommonModeCalculator();
     } else {
       cout << "Sorry Only Median is available for now, Mean and FastLinear are coming soon" << endl;
+      return;
     }
+  } else {
+    cout << "ApvAnalysisFactory: algorithm " << theAlgorithmType_ << " not supported" << endl:
```
Comment (perrotta) [inline CalibTracker/SiStripAPVAnalysis/src/ApvAnalysisFactory.cc:124]:
```suggestion
    cout << "ApvAnalysisFactory: algorithm " << theAlgorithmType_ << " not supported" << endl;
```

**Comment by iarspider** [comment]
please test

**Comment by perrotta** [comment]
+alca
- DAS error in the RelVal input test clearly uncorrelated

**Comment by antoniovilela** [comment]
+1

**Comment by antoniovilela** [comment]
ignore tests-rejected with external-failure

**Pull request merged**